### PR TITLE
Add a cache step

### PR DESCRIPTION
**Pull request by yozhgoor**

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@ jobs:
     outputs:
       size: ${{ steps.build.outputs.size }}
     steps:
+
+    - uses: Swatinem/rust-cache@v1
+
```
Comment (cecton) [inline .github/workflows/pr.yml:19]: The spacing looks different than the rest of the file

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -16,6 +16,8 @@ jobs:
     - uses: actions/checkout@v2
     - run: cargo test --all-features --verbose
 
+    - uses: Swatinem/rust-cache@v1
+
```
Comment (cecton) [inline .github/workflows/main.yml:20]: Same issue with the spacing

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -15,6 +15,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - run: cargo test --all-features --verbose
+    - uses: Swatinem/rust-cache@v1
```
Comment (cecton) [inline .github/workflows/main.yml:18]: If you trigger run the cache step AFTER the cargo command you will never restore anything lol

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -14,6 +14,7 @@ jobs:
     outputs:
       size: ${{ steps.build.outputs.size }}
     steps:
+    - uses: Swatinem/rust-cache@v1
     - uses: actions/checkout@v2
```
Comment (cecton) [inline .github/workflows/pr.yml:18]: Makes more sense to push the cache restore after the checkout

**Review by cecton — APPROVED** [review]
Fine :+1: You see that you need more practice XD these are beginner's mistake. Also you could have notice that the cache wasn't working if you tested. Here I think you added stuff and made the CI work but you didn't check the time it takes for the CI to complete, didn't you?

**Comment by yozhgoor** [comment]
> Fine 👍 You see that you need more practice XD these are beginner's mistake. Also you could have notice that the cache wasn't working if you tested. Here I think you added stuff and made the CI work but you didn't check the time it takes for the CI to complete, didn't you?

No, i haven't.
Is more logic now, I need stop just doing and figure out things before

I write it down in a corner!

**Pull request merged**